src/main/java/org/codehaus/plexus/util/xml/pull/MXParser.java

```diff
@@ -3656,7 +3656,8 @@ else if ( expand )
                 buf = newBuf;
                 if ( bufLoadFactor > 0 )
                 {
-                    bufSoftLimit = ( bufLoadFactor * buf.length ) / 100;
+                    // Include fix for https://web.archive.org/web/20070831191548/http://www.extreme.indiana.edu/bugzilla/show_bug.cgi?id=228
+                    bufSoftLimit = (int) ((((long) bufLoadFactor) * buf.length) / 100);
```

Comment:
if `bufLoadFactor` remains constant, then you can simplify:` bufSoftLimit = 2*bufSoftLimit;`

Reply:
> if `bufLoadFactor` remains constant, then you can simplify:` bufSoftLimit = 2*bufSoftLimit;`

I see what you mean because the only thing changing at the moment is the `buf.len` which is being doubled, so this calculation should essentially result in 2 times the current `bufSoftLimit`, but as we're not guaranteed that `bufSoftLimit` will remain constant (it's not declared as such) then airing on the side of accuracy over simplification seems to make sense here.

Reply:
It could be modified but it is very unlikely: the field is protected, with no setter and no constructor setting the value, which means only is modifiable by subclassing `MXParser`. Even it represents a percentage of the total storage so` (bufLoadFactor/100)` could be pre-computed. We could redefine `bufLoadFactor` as `protected float bufLoadFactor = 0.95f;`

Reply:
> It could be modified but it is very unlikely: the field is protected, with no setter and no constructor setting the value, which means only is modifiable by subclassing `MXParser`. This design is very leaky. Even it represents a percentage of the total storage so` (bufLoadFactor/100)` could be pre-computed. We could define a new field `protected float bufferLoadFactor = 0.95f;` and deprecate the 'bufLoadFactor' field.

Agree with your suggestion it helps simplify it https://github.com/codehaus-plexus/plexus-utils/pull/72/commits/6c55f731aa792973a60a82eb114658428892c26c